examples/generic/generic_systems.rs

```diff
@@ -0,0 +1,11 @@
+use bevy::ecs::component::Component;
+
+// remove all entites that aren't the player
+fn remove_entities<T: Component>(q: Query<Entity, With<T>>) {
+    for entity in q.iter() {
+        let id = Entity;
+        if entity != id.0 {
+            command.entity(entity).despawn();
+        }
+    }
```

Comment:
It's a little confusing as to what this loop is doing, and I don't believe there's any need to compare `entity != id.0`.
Each `entity` that comes in is an entity that exists with this component. 
this loop could be simplified to 
```rust
q.for_each(|entity| { commands.entity(entity).despawn(); });
```